plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java

```diff
@@ -33,13 +41,32 @@ public HudsonPrivateSecurityRealmConfigurator() {
                     .collect(Collectors.toList()))
             .setter((target, value) -> {
                 for (UserWithPassword user : value) {
-                    target.createAccount(user.id, user.password);
+                    if (user.password.startsWith(HASHED_PASSWORD_PREFIX) && jenkinsSupportsHashedPasswords()) {
+                        try {
+                            target.createAccountWithHashedPassword(user.id, user.password);
+                        } catch (IllegalArgumentException e) {
+                            logger.warn("Failed to create user with presumed hashed password", e);
+                            // fallback, just create the account as is
+                            target.createAccount(user.id, user.password);
+                        }
+                    } else {
+                        target.createAccount(user.id, user.password);
+                    }
                 }
             }
         ));
         return describe;
     }
 
+    static boolean jenkinsSupportsHashedPasswords() {
```

Comment:
shouldn't be needed anymore with core version bump

Reply:
OK, wasn't sure how much backwards compatibility support was desired for this project.

Reply:
Resolved that this is needed still.